src/shared/condition.c

```diff
@@ -450,6 +451,87 @@ static int condition_test_architecture(Condition *c, char **env) {
         return a == b;
 }
 
+#define DTCOMPAT_FILE "/sys/firmware/devicetree/base/compatible"
+static int condition_test_firmware_devicetree_compatible(const char *dtcompatarg) {
+        int err;
+        _cleanup_free_ char *dtcompat = NULL;
+        char *compat;
+        size_t size;
+
+        err = read_full_virtual_file(DTCOMPAT_FILE, &dtcompat, &size);
+        if (err < 0) {
+                /* if the path doens't exist it is incompatible */
+                if (err != -ENOENT)
+                        log_debug_errno(errno, "Failed to open() '%s', assuming machine is incompatible: %m", DTCOMPAT_FILE);
+                return false;
+        }
+
+        /* Not sure this can happen, but play safe. */
+        if (!size) {
```

Comment:
we usually use C's downgrade to bool feature only for variables that actually are bools or at least boolean like – and for pointers, but never for values that are actually numbers, like this one. Hence please write as `size == 0`

Reply:
not addressed

Reply:
Oh, indeed, I failed to fix this. Fixed in the most recent push.